src/app/helpers/elements.ts

```diff
@@ -72,3 +75,19 @@ export const sort = (elements: Element[], sort: Sort, labelID: string) => {
     };
     return [...elements].sort((e1, e2) => compare(e1, e2));
 };
+
+export const getCounterMap = (
+    labels: Label[],
+    conversationCounters: LabelCount[],
+    messageCounters: LabelCount[],
+    mailSettings: MailSettings
+) => {
+    const result: { [labelID: string]: LabelCount | undefined } = {};
+    const labelIDs = [...Object.values(MAILBOX_LABEL_IDS), ...labels.map((label) => label.ID || '')];
+    labelIDs.forEach((labelID) => {
+        const conversationMode = isConversationMode(labelID, mailSettings);
+        const counters = conversationMode ? conversationCounters : messageCounters;
+        result[labelID] = counters.find((counter) => counter.LabelID === labelID);
+    });
+    return result;
```

Comment:
- What do you think about use a `reduce` to create the `result` variable?
- To make the research faster, create a map first and use it to find `counter`.

``` jsx
import { toMap } from 'proton-shared/lib/helpers/object';

const conversationCountersMap = toMap(conversationCounters, 'LabelID');
const messageCountersMap = toMap(messageCounters, 'LabelID');

return labelIDs.reduce((acc, labelID) => {
        const conversationMode = isConversationMode(labelID, mailSettings);
        const countersMap = conversationMode ? conversationCountersMap : messageCountersMap;

        acc[labelID] = countersMap[labelID];

        return acc;
    }, { [labelID: string]: LabelCount | undefined } = {});
```

Reply:
The reduce is more "functionnal" but doesn't change much for populating an object but why not.

About the counters map, you're right, using `find` inside a loop is pretty bad.